### PR TITLE
Rework filter bots macro

### PR DESCRIPTION
**Pull request by agnessnowplow**
## Description

This PR makes sensible adjustments in the filter_bots() macro based on user request.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
https://snplow.atlassian.net/browse/PE-6533

## Checklist

- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: models/optional_modules/core_web_vitals/scratch/default/snowplow_unified_web_vital_events_this_run.sql [hunk 1]
```diff
@@ -69,10 +69,6 @@ with prep as (
 
   -- exclude bot traffic
 
-  {% if var('snowplow__enable_iab', false) %}
-    and not e.iab__spider_or_robot = True
```
Comment (ilias1111) [inline models/optional_modules/core_web_vitals/scratch/default/snowplow_unified_web_vital_events_this_run.sql:73]: We don't we go with

`and not e.iab__spider_or_robot` ? Not important ofc, but its redundant

File: macros/filter_bots.sql [hunk 2]
```diff
@@ -10,17 +10,31 @@ You may obtain a copy of the Snowplow Personal and Academic License Version 1.0
 {%- endmacro -%}
 
 {% macro default__filter_bots(table_alias = none) %}
+  {% if var('snowplow__enable_iab', false) %}
+    and (not iab__spider_or_robot = True
```
Comment (ilias1111) [inline macros/filter_bots.sql:14]: How about something like
`and coalesce(iab__spider_or_robot, False ) = False`

File: models/optional_modules/core_web_vitals/scratch/default/snowplow_unified_web_vital_events_this_run.sql (hunk 1 above)
Comment (ilias1111) [reply models/optional_modules/core_web_vitals/scratch/default/snowplow_unified_web_vital_events_this_run.sql:73]: Or at least e.iab__spider_or_robot = False, so we skip the not, and we mention what matters ( what we keep ) .

Though fair point is we are creating a macro to exclude so what we dont keep also makes sense to be mentioned

File: macros/filter_bots.sql (hunk 2 above)
Comment (agnessnowplow) [reply macros/filter_bots.sql:14]: Sure, I like this option, cleaner as well as easier on the evaluation side due to removing or.

**Pull request merged**